### PR TITLE
feat-added protected route functionality

### LINKED ISSUE #10: Restrict some routes for unauthorised users
Currently, all the users can navigate through all the routes, they don't need to authenticate. We need to restrict users from navigating to protected routes. 

Features to implement: 
* Authenticate the user to and save the token
* Then, if the token is present then show protected route otherwise users will be redirected to the login page.

### PR DESCRIPTION
**Pull request by manish591**
This fixes #10

### CODE REVIEW

**Review by shrey27 — APPROVED** [review]
Your use case of Private Route is very, versatile and clean. It can be very useful when handling large scale use cases involving private routes.

**Review by abhisheksunil2201 — APPROVED** [review]
I like the use of protected route you have implemented here 👍🏻

File: src/components/index.js [hunk 1]
```diff
@@ -6,4 +6,5 @@ export { Footer } from './footer/Footer';
 export { ProductCardVertical } from './product-card/ProductCardVertical';
 export { ProductCardHorizontal } from './product-card/ProductCardHorizontal';
 export { Loader } from './loader/Loader';
-export { ProfileDropdown } from './profile-dropdown/Profile.dropdown';
\ No newline at end of file
+export { ProfileDropdown } from './profile-dropdown/Profile.dropdown';
+export { ProtectedRoute } from './protected-route/ProtectedRoute';
```
Comment (varnita619) [inline src/components/index.js:10]: you can add new line at the end of each file just for cleanliness

**Review by varnita619 — APPROVED** [review]
Folder structure and code is clean

**Pull request merged**